zulipterminal/helper.py

```diff
@@ -491,23 +492,25 @@ def match_stream(data: List[Tuple[DataT, str]], search_text: str,
     matches = OrderedDict([
         ('pinned', defaultdict(list)),
         ('unpinned', defaultdict(list)),
-    ])  # type: OrderedDict[str, DefaultDict[int, List[DataT]]]
+    ])  # type: OrderedDict[str, DefaultDict[int, List[Tuple[DataT, str]]]]
 
     for datum, splits in stream_splits:
         kind = 'pinned' if splits[0] in pinned_stream_names else 'unpinned'
         for match_position, word in enumerate(splits):
             if word.lower().startswith(search_text.lower()):
-                matches[kind][match_position].append(datum)
+                matches[kind][match_position].append((datum, splits[0]))
```

Comment:
As `splits[0]` has two occurrences now (line 498 and line 501), it might be clearer to extract it as a variable - `stream_name`.

Reply:
Updated - thanks for the review :+1: